### PR TITLE
Default deb config should work with default istio.yaml

### PR DESCRIPTION
**Pull request by costinm**
Current defaults are valid for the istio-auth.yaml

For 0.6 we should make 15003 use mTLS (if CA exists), to allow gradual adoption and opt-in, 
with some workloads using 8080/insecure and some using 15003 mTLS. We can't have all or nothing, 
and SNI sniffing is not yet ready.

### CODE REVIEW

File: tools/deb/istio-start.sh [hunk 1]
```diff
@@ -39,8 +39,11 @@ ISTIO_CFG=${ISTIO_CFG:-/var/lib/istio}
 NS=${ISTIO_NAMESPACE:-default}
 SVC=${ISTIO_SERVICE:-rawvm}
 ISTIO_SYSTEM_NAMESPACE=${ISTIO_SYSTEM_NAMESPACE:-istio-system}
-ISTIO_PILOT_PORT=${ISTIO_PILOT_PORT:-15003}
-CONTROL_PLANE_AUTH_POLICY=${CONTROL_PLANE_AUTH_POLICY:-MUTUAL_TLS}
+
+# The default matches the default istio.yaml - use sidecar.env to override this if you
```
Comment (ldemailly) [inline tools/deb/istio-start.sh]: I'm torn about that one, I think auth (smart enough to not break things) should be the default - specially for mesh expansion where we want to join securely

File: tools/deb/sidecar.env [hunk 2]
```diff
@@ -37,7 +40,7 @@
 # ENVOY_USER=istio-proxy
 
 # Uncomment to enable debugging
-# ISTIO_DEBUG="-l debug"
+# AGENT_FLAGS="--proxyLogLevel debug"
```
Comment (ldemailly) [inline tools/deb/sidecar.env]: ISTIO_AGENT_FLAGS ?

File: tools/deb/istio-start.sh [hunk 3]
```diff
@@ -39,8 +39,11 @@ ISTIO_CFG=${ISTIO_CFG:-/var/lib/istio}
...
+# enable auth. This requires node-agent to be running.
+ISTIO_PILOT_PORT=${ISTIO_PILOT_PORT:-8080}
+CONTROL_PLANE_AUTH_POLICY=${CONTROL_PLANE_AUTH_POLICY:-NONE}
```
Comment (ldemailly) [inline tools/deb/istio-start.sh]: it's quite longish but this should probably have been ISTIO_CONTROL_PLANE_AUTH when first added

File: tools/deb/istio-start.sh (hunk 1 above)
Comment (costinm) [reply tools/deb/istio-start.sh]: Agree - however it is not the default today in istio.yaml.

See my comment - for 0.6 we should make 15003 mTLS by default. We still need to migrate,
and support cases where people don't do it all at once.

File: tools/deb/istio-start.sh (hunk 3 above)
Comment (costinm) [reply tools/deb/istio-start.sh]: Just used what the script was using - agree, we can tweak it later.

File: tools/deb/sidecar.env (hunk 2 above)
Comment (costinm) [reply tools/deb/sidecar.env]: Done.

File: tools/deb/istio-start.sh (hunk 3 above)
Comment (costinm) [reply tools/deb/istio-start.sh]: Updated to ISTIO_CP_AUTH

**Pull request merged**